src/System.Management.Automation/engine/ParameterBinderBase.cs

```diff
@@ -787,12 +787,18 @@ internal CommandLineParameters CommandLineParameters
 
             // Ensure that each element abides by the metadata
             bool isEmpty = true;
+            Type elementType = parameterMetadata.CollectionTypeInformation.ElementType;
+            bool isElementValueType = elementType != null && elementType.IsValueType;
             // Note - we explicitly don't pass the context here because we don't want
             // the overhead of the calls that check for stopping.
             while (ParserOps.MoveNext(null, null, ienum))
             {
-                object element = ParserOps.Current(null, ienum);
                 isEmpty = false;
+                // If the element of the collection is of value type, then no need to check for null
+                // because a value-type value cannot be null.
+                if (isElementValueType) { break; }
```

Comment:
Why if we calculate this before the cycle?

Reply:
Same reason as above. The element type is resolved using the collection type.